### PR TITLE
services.ec2: Raise exceptions when shell execution fails

### PR DESCRIPTION
**Pull request by sayanchowdhury**
Signed-off-by: Sayan Chowdhury <sayan.chowdhury2012@gmail.com>

### CODE REVIEW

File: fedimg/utils.py [hunk 1]
```diff
@@ -99,13 +101,15 @@ def get_value_from_dict(_dict, *keys):
...
+    _log.info("Starting the command: %r" % command)
+    ret = subprocess.Popen(' '.join(command), stdin=subprocess.PIPE,
+                           shell=True, stdout=subprocess.PIPE,
+                           stderr=subprocess.PIPE, close_fds=True)
     out, err = ret.communicate()
-    _log.debug("Finished executing the command: %r" % command)
     retcode = ret.returncode
+    _log.info("Finished executing the command: %r" % command)
+    _log.debug("(output) %s, (error) %s, (retcode) %s" % (
+               out, err, retcode))
```
Comment (dustymabe) [inline fedimg/utils.py]: i would say this line should get output any time the returncode != 0

File: fedimg/utils.py [hunk 2]
```diff
@@ -128,17 +132,27 @@ def unxz_source_file(file_path):
     ])
 
     if retcode != 0:
-        return ''
+        _log.error("Unable to unxz the image: (output) %s,"
+                   "(error) %s, (return_code) %s" % (
+                       output, error, retcode))
```
Comment (dustymabe) [inline fedimg/utils.py]: The above can be simplified if we just print the error output above in the external_run_command function

File: fedimg/utils.py [hunk 3]
```diff
@@ -128,17 +132,27 @@ def unxz_source_file(file_path):
...
+        raise CommandRunFailed
 
-    return file_path.rstrip('.xz')
+    raw_file_path = file_path.rstrip('.xz')
+    if not os.path.isfile(raw_file_path):
+        _log.error("The requested file: %r does not exists."
+                   "(output) %s, (error) %s, (return_code) %s" % (
```
Comment (dustymabe) [inline fedimg/utils.py]: we don't need to pint output error return_code again

File: fedimg/utils.py [hunk 4]
```diff
@@ -147,7 +161,16 @@ def get_source_from_image(image_url):
     ])
 
     if retcode != 0:
-        return ''
+        _log.error("Unable to download the image: (output) %s,"
+                   "(error) %s, (return_code) %s" % (
+                       output, error, retcode))
```
Comment (dustymabe) [inline fedimg/utils.py]: same comment as above

File: fedimg/utils.py [hunk 5]
```diff
@@ -147,7 +161,16 @@ def get_source_from_image(image_url):
...
+        raise CommandRunFailed
+
+    if not os.path.isfile(file_path):
+        _log.error("The requested file: %r does not exists."
+                   "(output) %s, (error) %s, (return_code) %s" % (
```
Comment (dustymabe) [inline fedimg/utils.py]: same comment as above

File: fedimg/utils.py (hunk 1 above)
Comment (sayanchowdhury) [reply fedimg/utils.py]: It would always log the output in `DEBUG` level.

**Comment by sayanchowdhury** [comment]
I've updated the PR @dustymabe. Can re-review?

File: fedimg/utils.py (hunk 1 above)
Comment (dustymabe) [reply fedimg/utils.py]: right, but is that level set by default?

Comment (dustymabe) [reply fedimg/utils.py]: i'm thinking it should be
```
if ret.returncode != 0:
     _log.error("command failed during run") 
     _log.error("(output) %s, (error) %s, (retcode) %s" % (out, err, retcode))
```

Comment (sayanchowdhury) [reply fedimg/utils.py]: Updated them. Good to merge?

File: fedimg/utils.py [hunk 6]
```diff
@@ -99,13 +101,18 @@ def get_value_from_dict(_dict, *keys):
...
+                           stderr=subprocess.PIPE, close_fds=True)
     out, err = ret.communicate()
-    _log.debug("Finished executing the command: %r" % command)
     retcode = ret.returncode
+    _log.info("Finished executing the command: %r" % command)
+
+    if retcode != 0:
+        _log.error("Command failed during run")
+        _log.error("(output) %s, (error) %s, (retcode) %s" % (out, err, retcode))
+
```
Comment (dustymabe) [inline fedimg/utils.py]: this looks good. but let's also have an unconditional: `_log.debug("(output) %s, (error) %s, (retcode) %s" % (out, err, retcode))`

Comment (dustymabe) [reply fedimg/utils.py]: or it can even be part of `else:`

Comment (sayanchowdhury) [reply fedimg/utils.py]: `else:` sounds better!

**Comment by dustymabe** [comment]
LGTM

**Pull request merged**